packages/glib/build.sh

```diff
@@ -67,11 +66,62 @@ termux_step_host_build() {
 termux_step_pre_configure() {
 	# glib checks for __BIONIC__ instead of __ANDROID__:
 	CFLAGS+=" -D__BIONIC__=1"
+	_PREFIX=$TERMUX_PKG_TMPDIR/prefix
+	local _WRAPPER_BIN="${TERMUX_PKG_BUILDDIR}/_wrapper/bin"
+	rm -rf $_PREFIX $_WRAPPER_BIN
+	mkdir -p $_PREFIX $_WRAPPER_BIN
 
-	TERMUX_PKG_VERSION=. termux_setup_gir
+	sed '/^export PKG_CONFIG_LIBDIR=/s|$|:'${_PREFIX}'/lib/pkgconfig|' \
+		"${TERMUX_STANDALONE_TOOLCHAIN}/bin/pkg-config" \
+		> "${_WRAPPER_BIN}/pkg-config"
+	chmod +x "${_WRAPPER_BIN}/pkg-config"
+	export PKG_CONFIG="${_WRAPPER_BIN}/pkg-config"
+	export PATH="${_WRAPPER_BIN}:${PATH}"
+
+	# Magic happens here.
+	# I borrowed nested building method from https://github.com/termux/termux-packages/blob/1244c75380beefc7f7da9744d55aa88df1640acb/x11-packages/qbittorrent/build.sh#L21-L28
+	# and modified termux_step_configure_meson in runtime to make it use another prefix
+	# Also I used advice from here https://github.com/termux/termux-packages/issues/20447#issuecomment-2156066062
+
+	(
+		eval `declare -f termux_step_configure_meson | sed 's|\$TERMUX_PREFIX|'$_PREFIX'|g'`
+		TERMUX_PKG_BUILDDIR=$TERMUX_PKG_TMPDIR/glib-build
+		mkdir -p $TERMUX_PKG_BUILDDIR
+		TERMUX_PKG_EXTRA_CONFIGURE_ARGS="${TERMUX_PKG_EXTRA_CONFIGURE_ARGS/"-Dintrospection=enabled"/"-Dintrospection=disabled"}"
+		TERMUX_PKG_VERSION=. termux_setup_gir
+		
+		cd "$TERMUX_PKG_BUILDDIR"
+		termux_step_configure
+		cd "$TERMUX_PKG_BUILDDIR"
+		termux_step_make
+		cd "$TERMUX_PKG_BUILDDIR"
+		termux_step_make_install
+	)
+
+	(
+		eval `declare -f termux_step_configure_meson | sed 's|\$TERMUX_PREFIX|'$_PREFIX'|g'`
+		TERMUX_PKG_BUILDER_DIR=$TERMUX_SCRIPTDIR/packages/gobject-introspection
+		TERMUX_PKG_BUILDDIR=$TERMUX_PKG_TMPDIR/gobject-introspection-build
+		TERMUX_PKG_SRCDIR=$TERMUX_PKG_TMPDIR/gobject-introspection-src
+		mkdir -p $TERMUX_PKG_BUILDDIR $TERMUX_PKG_SRCDIR
+		. $TERMUX_PKG_BUILDER_DIR/build.sh
+		cd "$TERMUX_PKG_CACHEDIR"
+
+		termux_step_get_source
+		termux_step_get_dependencies_python
+		termux_step_patch_package
+
+		termux_step_pre_configure
 
-	# Workaround: Remove cyclic dependency between gir and glib
-	sed -i "/Requires:/d" "${TERMUX_PREFIX}/lib/pkgconfig/gobject-introspection-1.0.pc"
+		cd "$TERMUX_PKG_BUILDDIR"
+		termux_step_configure
+		cd "$TERMUX_PKG_BUILDDIR"
+		termux_step_make
+		cd "$TERMUX_PKG_BUILDDIR"
+		termux_step_make_install
+	)
+
+	TERMUX_PKG_VERSION=. termux_setup_gir
```

Comment:
This could use a comment.
What the hell is going on here?

I do recall seeing this in other build scripts that need `gir`,
but I think a comment would really help here.

Reply:
`termux_setup_gir` is used in more than 20 packages and does not have comments.

Reply:
It is unclear to me from looking at it what `TERMUX_PKG_VERSION=. termux_setup_gir` means.
Obviously we're setting up GIR, that much is obvious.
But why are we passing `TERMUX_PKG_VERSION=.` to it?
We don't pass variable overwrites to other setup functions.

Reply:
If I am reading this right it determines the location the generated files end up in.
Which is extremely non-obvious considering we're passing an override to the package version.
https://github.com/termux/termux-packages/blob/1244c75380beefc7f7da9744d55aa88df1640acb/scripts/build/setup/termux_setup_gir.sh#L27-L37

Reply:
Something to the effect of:
```bash
	# Place the GIR files inside the root of the GIR directory (gir/.) of the package
	TERMUX_PKG_VERSION=. termux_setup_gir
```
Would go a long way.
If you come up with better wording, by all means.

Reply:
It was my invention to make packages using `termux_setup_gir` auto-updatable.

`termux_setup_gir` uses `TERMUX_PKG_VERSION` to build path to folder containing gir xmls, so I am using `TERMUX_PKG_VERSION=.` to make it version independent. 

Compare [gegl](https://github.com/termux/termux-packages/tree/1244c75380beefc7f7da9744d55aa88df1640acb/packages/gegl) (does not use `TERMUX_PKG_VERSION=.` and [libgsf](https://github.com/termux/termux-packages/tree/1244c75380beefc7f7da9744d55aa88df1640acb/packages/libgsf) (which uses `TERMUX_PKG_VERSION=.`).

Reply:
Oh, github did not show new review comments. Ok.

Reply:
> It was my invention to make packages using `termux_setup_gir` auto-updatable.

Unfortunately, I'm not in your head.

> `termux_setup_gir` uses `TERMUX_PKG_VERSION` to build path to folder containing gir xmls, so I am using `TERMUX_PKG_VERSION=.` to make it version independent.

I have only a very vague understanding of how GIR works.
So any quirks like that would be very welcome to be explained when they crop up.

> Compare [gegl](https://github.com/termux/termux-packages/tree/1244c75380beefc7f7da9744d55aa88df1640acb/packages/gegl) (does not use `TERMUX_PKG_VERSION=.` and [libgsf](https://github.com/termux/termux-packages/tree/1244c75380beefc7f7da9744d55aa88df1640acb/packages/libgsf) (which uses `TERMUX_PKG_VERSION=.`).

Again that does make sense after digging through the code,
but a build script should be explained without needing to check the source for every utility function used in it.
GIR and package version do not form an association with directory structure in my mind.

Reply:
Probably it will be better to make top-level variable like `TERMUX_PKG_GIR_UNVERSIONED` and remove all invocations of `TERMUX_PKG_VERSION=.` in separate commit.

Reply:
Sounds good to me.